mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,301 @@
+# Authors: Kyle Mathewson, Jonathan Kuziek <kuziek@ualberta.ca>
+#
+# License: BSD (3-clause)
+
+import re as re
+
+import numpy as np
+
+from ..base import BaseRaw
+from ..meas_info import create_info
+from ...utils import logger, verbose, fill_doc
+from ...annotations import Annotations
+
+
+@fill_doc
+def read_raw_boxy(fname, datatype='AC', preload=False, verbose=None):
+    """Reader for a BOXY optical imaging recording.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the BOXY data file.
+    datatype : str
+        Type of data to return (AC, DC, or Ph).
+    %(preload)s
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of RawBOXY
+        A Raw object containing BOXY data.
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    return RawBOXY(fname, datatype, preload, verbose)
+
+
+@fill_doc
+class RawBOXY(BaseRaw):
+    """Raw object from a BOXY optical imaging file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the BOXY data file.
+    datatype : str
+        Type of data to return (AC, DC, or Ph).
+    %(preload)s
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, datatype='AC', preload=False, verbose=None):
+        logger.info('Loading %s' % fname)
+
+        # Determine which data type to return.
+        if datatype not in ['AC', 'DC', 'Ph']:
+            raise RuntimeError('Expect AC, DC, or Ph, got %s' % datatype)
+
+        # Read header file and grab some info.
+        filetype = 'parsed'
+        start_line = 0
+        end_line = 0
+        mrk_col = 0
+        mrk_data = list()
+        col_names = list()
+        with open(fname, 'r') as data:
+            for line_num, i_line in enumerate(data, 1):
+                if '#DATA ENDS' in i_line:
+                    # Data ends just before this.
+                    end_line = line_num - 1
+                    break
+                if 'Detector Channels' in i_line:
+                    detect_num = int(i_line.rsplit(' ')[0])
+                elif 'External MUX Channels' in i_line:
+                    source_num = int(i_line.rsplit(' ')[0])
+                elif 'Update Rate (Hz)' in i_line:
+                    srate = float(i_line.rsplit(' ')[0])
+                elif 'Updata Rate (Hz)' in i_line:
```

Comment:
Typo or this is a thing?

Reply:
typo

Reply:
sort of both, some older versions of BOXY saved 'Update Rate' as 'Updata Rate'. So I'm thinking we might want to keep this in for compatibility reasons.

@kylemath the 'anc' and 'emm' datasets have this typo, if you want to take a look.

Reply:
Ok if its a genuine "bug" in the file format saved by boxy then keep it in so that we can provide support for different file format versions. But please add a comment explaining this like `# In boxy versions 77-81 the vendor software exported values with the type Updata`

Reply:
Okay, will do.